src/gui/tray/activitydata.cpp

```diff
@@ -82,10 +98,15 @@ OCC::Activity Activity::fromActivityJson(const QJsonObject &json, const AccountP
                 parameterJsonObject.value(QStringLiteral("id")).toString(),
                 parameterJsonObject.value(QStringLiteral("name")).toString(),
                 parameterJsonObject.contains(QStringLiteral("path")) ? parameterJsonObject.value(QStringLiteral("path")).toString() : QString(),
-                parameterJsonObject.contains(QStringLiteral("link")) ? QUrl(parameterJsonObject.value(QStringLiteral("link")).toString()) : QUrl(),
+                parameterJsonObject.contains(QStringLiteral("link")) ? stringToUrl(parameterJsonObject.value(QStringLiteral("link")).toString()) : QUrl(),
             };
+
+            if (activity._objectType == "calendar") {
+                activity._link = activity._subjectRichParameters[i.key()].link;
```

Comment:
This errors if there is a `object_type=user` entry in the subjectRichParameters array which is the case when someone else performs the action? do you need to do some kind of `isset` or `not null` check on `.link` first?

Additionally to `activity._objectType == "calendar"` I would recommend to **not** set the activity_link when it was set already (so that the link provided by the activity wins)

Reply:
> This errors if there is a `object_type=user` entry in the subjectRichParameters array which is the case when someone else performs the action? do you need to do some kind of `isset` or `not null` check on `.link` first?

No, it does not. 

> Additionally to activity._objectType == "calendar" I would recommend to not set the activity_link when it was set already (so that the link provided by the activity wins)

I have not seen this case yet, that is why I asked previously if 'link' could always be used instead of having it in subjectRichParameters sometimes.

Reply:
> do you need to do some kind of `isset` or `not null` check on `.link` first

I am checking now if it is empty before assigning it.

Reply:
we may be currently limited in our ability to provide desktop native notifications with rich text on all our platforms
clickable links would not work on macOS for example